slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -2666,6 +2677,129 @@ uint32_t slsDetector::getADCEnableMask() {
+    FILE_LOG(logDEBUG1) << "Setting External Sampling Source to " << arg;
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_EXTERNAL_SAMPLING_SOURCE, arg, retval);
+        FILE_LOG(logDEBUG1) << "External Sampling source: " << retval;
+    }
+    return retval;
+}
+
+int slsDetector::getExternalSamplingSource() {
+    return setExternalSamplingSource(-1);
+}
+
+int slsDetector::setExternalSampling(int value) {
+    int arg = value;
+    int retval = -1;
+    FILE_LOG(logDEBUG1) << "Setting External Sampling to " << arg;
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_EXTERNAL_SAMPLING, arg, retval);
+        FILE_LOG(logDEBUG1) << "External Sampling: " << retval;
+    }
+    return retval;
+}
+
+int slsDetector::getExternalSampling() {
+    return setExternalSampling(-1);
+}
+
+void slsDetector::setReceiverDbitList(std::vector<int> list) {
+    FILE_LOG(logDEBUG1) << "Setting Receiver Dbit List";
+
+    if (list.size() > 64) {
+        throw sls::RuntimeError("Dbit list size cannot be greater than 64\n");
+    }
+    for (auto &it : list) {
+        if (it < 0 || it > 63) {
+            throw sls::RuntimeError("Dbit list value must be between 0 and 63\n");
+        }
+    }
+
+    // copy size and vector to shm
+    shm()->rxDbitListSize = list.size();
+    std::copy(list.begin(), list.end(), shm()->rxDbitList);
+ 
+    if (shm()->rxOnlineFlag == ONLINE_FLAG) {
+        int args[list.size() + 1];
+        args[0] = list.size();
+        std::copy(std::begin(list), std::end(list), args + 1);
+        sendToReceiver(F_SET_RECEIVER_DBIT_LIST, args, sizeof(args), nullptr, 0);
+    }
+}
+
+std::vector<int> slsDetector::getReceiverDbitList() {
+    int fnum = F_GET_RECEIVER_DBIT_LIST;
+    int ret = FAIL;
+    std::vector <int> retval;
+    int retsize = 0;
+    FILE_LOG(logDEBUG1) << "Getting Receiver Dbit List";
+    if (shm()->rxOnlineFlag == ONLINE_FLAG) {
+        auto receiver =
+        sls::ClientSocket("Receiver", shm()->rxHostname, shm()->rxTCPPort);
+        receiver.sendData(&fnum, sizeof(fnum));
+        receiver.receiveData(&ret, sizeof(ret));
+        if (ret == FAIL) {
+            char mess[MAX_STR_LENGTH]{};
+            receiver.receiveData(mess, MAX_STR_LENGTH);
+            throw ReceiverError("Receiver " + std::to_string(detId) +
+                            " returned error: " + std::string(mess));
+        }
+        receiver.receiveData(&retsize, sizeof(retsize));
+        int list[retsize];
+        receiver.receiveData(list, sizeof(list));
+
+        // copy after no errors
+        shm()->rxDbitListSize = retsize;
+        std::copy(list, list + retsize, shm()->rxDbitList);
+    } 
+
+    if (shm()->rxDbitListSize) {
+        retval.resize(shm()->rxDbitListSize);
+        std::copy(shm()->rxDbitList, shm()->rxDbitList + shm()->rxDbitListSize, std::begin(retval));
+    }
+
+    return retval;
+}
+
+int slsDetector::setReceiverDbitOffset(int value) {
+    int arg = value;
```

Comment:
No need to copy value into arg. You can use value to send it directly
sendToReceiver(F_RECEIVER_DBIT_OFFSET, value, retval);